keras/backend/tensorflow/core.py

```diff
@@ -28,7 +28,10 @@ def handle(self):
 
     def _initialize(self, value):
         self._value = tf.Variable(
-            value, dtype=self._dtype, trainable=self.trainable, name=self.name
+            tf.cast(value, self._dtype),
```

Comment:
Why is the cast necessary? Wouldn't the `tf.Variable` always be `self._dtype` regardless of the dtype of `value`?

Reply:
The cast was necessary due to the canonicalization, but now it can be removed.
Done.